components/MyCourses/MyCourses.js

```diff
@@ -0,0 +1,140 @@
+import {
+  Card,
+  CardContent,
+  CardMedia,
+  CardActionArea,
+  Typography,
+  Box,
+  Stack,
+  List,
+  ListItem,
+  Dialog,
+  DialogTitle,
+  DialogActions,
+  DialogContent,
+  DialogContentText,
+  TextField
+} from '@mui/material';
+import AdapterDateFns from '@mui/lab/AdapterDateFns';
+import LocalizationProvider from '@mui/lab/LocalizationProvider';
+import DateTimePicker from '@mui/lab/DateTimePicker';
+import EditIcon from '@mui/icons-material/Edit';
+import { useState } from 'react';
+import MainButton from '../basecomponents/MainButton.js'
+
+export default function MyCourses({ course }) {
+  const [showStudentList, setShowStudentList] = useState(false);
+  const [editCourseInfo, setEditCourseInfo] = useState(false);
+  const [userType, setUserType] = useState('mentor');
+  const [currCourseName, setCurrCourseName] = useState(course.courseName);
+  const [currStartTime, setCurrStartTime] = useState(course.startTime);
+  const [currEndTime, setCurrEndTime] = useState(course.endTime);
+  const [newCourseName, setNewCourseName] = useState('');
+  const [newStartTime, setNewStartTime] = useState(null);
+  const [newEndTime, setNewEndTime] = useState(null);
+
+  const handleStudentList = () => {
+    if (userType === 'mentor') {
+      setShowStudentList(!showStudentList)
```

Comment:
Instead of using '!' sign, just make this function set state as true. Less ambiguity and wont risk falling out of state.

Reply:
@cmjosephs Hey Chris! It's meant to toggle so that you can open and close the student list on each class card.  I can change it and write another function to close the modal if you'd like though.